### ISSUE TITLE
Unable to verify phone number

### ISSUE THREAD

**Issue by sebastiannielsen** [keybase/keybase-issues#3502]
Unable to verify phone number
ERROR CODE 218 - Failed to send SMS Bad HTTP Code 400; Message to number +46313107835, is not a mobile number, more info: https://www.twilio.com/docs/errors/21613 details= in method keybase.1.phoneNumbers.resendVerificationForPhoneNumber

Shouldn't it call if it is unable to send SMS?
Have a "fixed line number" connected to my SIM card.

https://www.telia.se/privat/telefoni/tjanster/produkt/mitt-hemnummer

**Comment by mjohngreene** [comment]
Same.  Cell # is definitely correct although no message ever arrives on phone.  Tried resending multiple times but never received message with verification code.

**Comment by heronhaye** [comment]
@mjohngreene What country are you in? Anything special about your phone plan?

**Comment by mjohngreene** [comment]
I'm in the US.  My plan is with Republic Wireless.  I'm not sure it is really "special" or anything but Republic phones use wifi when available (ie, it automatically switches to wifi if I am in range of one I have access to) to keep cell costs down.

**Comment by plttn** [comment]
Just covering some bases: looks like @sebastiannielsen's number isn't considered a mobile number as far as Twilio is concerned. My guess is it codes as a home phone number which means Twilio won't use it. 

@mjohngreene's number looks like the problem that Google Voice has. Since Republic Wireless uses Bandwidth.com as the backing carrier, that's considered a VoIP number, so Twilio is silently failing on that number too.

**Comment by sebastiannielsen** [comment]
I know that twilio has the possibility to call as a fallback if SMS fails, I think keybase should have that option. Ergo it calls a normal phone call to the number and reads the code aloud as a voice if SMS sending fails.

**Comment by Lucky225** [comment]
> Just covering some bases: looks like @sebastiannielsen's number isn't considered a mobile number as far as Twilio is concerned. My guess is it codes as a home phone number which means Twilio won't use it.
> 
> @mjohngreene's number looks like the problem that Google Voice has. Since Republic Wireless uses Bandwidth.com as the backing carrier, that's considered a VoIP number, so Twilio is silently failing on that number too.

Shouldn't. Twilio uses netnumber, so even if it shows as 'untextable' they can see it's loaded on netnumber. I've verified my Google Voice and another VoIP number without issue

**Comment by mjohngreene** [comment]
Never mind.  The texts were apparently coming through but not appearing/rendering on my phone.  So, it looks as though it was a problem with my phone and not Keybase's verification process.  Apologies for the head fake.

**Comment by fbay26** [comment]
@mjohngreene i have the same issue from Turkey. Could you find a solution?

Thanks.

**Comment by heronhaye** [comment]
@fbay26 what kind of carrier are you using? should work from turkey.

**Comment by fbay26** [comment]
@heronhaye TurkTelekom

**Comment by heronhaye** [comment]
Do you have any issues with getting verification codes from other services?

**Comment by heronhaye** [comment]
@fbay26 Looks like your phone number is verified. Does your app tell you otherwise?

**Comment by fbay26** [comment]
Probably its related with only TurkTelekom, i got verification sms to other service Vodafone

**Comment by longnd** [comment]
I'm from Vietnam, using Viettel as the carrier but also can't receive the OTP SMS to verify my phone number.

**Comment by heronhaye** [comment]
@longnd Is your Keybase username `longnd`? According to our SMS provider, your verification code was sent successfully. Maybe it was blocked on your carrier's side?

**Comment by longnd** [comment]
@heronhaye yes, that's my username. I just managed to verify my phone number BUT it's weird. Today I tried it several times, there were 2 times the SMS were received but a long time after I requested the OTP so when I enter it, it was expired already. The other times, the SMS never come. But my last try, it comes right after I request it. My carrier doesn't block it, I believe.

**Comment by umiemnieumiem** [comment]
@heronhaye hello, I have same issue in Poland, Plus provider on Android device.

**Comment by heronhaye** [comment]
@longnd It should say something like that. Did that code in your screenshot work?

@umiemnieumiem Please be more specific and include your Keybase username.

**Comment by umiemnieumiem** [comment]
@heronhaye my username wiemniewiem on Keybase. Despite removing and adding again my number, resending SMS multiple times or waiting patiently for one, no messege has ever appeared on my phone

**Comment by heronhaye** [comment]
@umiemnieumiem Sorry, it looks to be all good on our side. Maybe your carrier is blocking the texts.

**Comment by longnd** [comment]
> @longnd It should say something like that. Did that code in your screenshot work?
@heronhaye yes, it works.

**Comment by seobaka** [comment]
@heronhaye Hi, I have same problem in Moldova, provider Idc.md. My keybase login is lexfiero

**Comment by heronhaye** [comment]
@seobaka According to our SMS provider, the number you provided is not a mobile phone number.

**Comment by openartstudio** [comment]
@heronhaye hello, I have same issue in Poland, Plus provider on Android device.

**Comment by seobaka** [comment]
@heronhaye but it is mobile number and other services like gmail, steam etc.  Sends sms on my number. How can i verify it in keybase?

**Comment by dvtkrlbs** [comment]
I dont get any sms neither. I am from Turkey and my carrier is Turk Telekom

**Comment by dvtkrlbs** [comment]
Hmm i tried the verification from my phone (instead of my laptop) through a vpn and i got the sms.

**Comment by steevepay** [comment]
Hi, it's impossible to verify my phone number.
Username: steevepay 
Country: France (+33)

**Comment by heronhaye** [comment]
@steevepay Sorry, all looks good on our side. Contact your carrier.

**Comment by Kabuk21** [comment]
Hi. O dont receive the SMS.
Username: Virgo85
Country: Malta +356

**Comment by steevepay** [comment]
> @steevepay Sorry, all looks good on our side. Contact your carrier.

I have just succeed by restarting the smartphone, it has never happened before 🤷‍♂️ thanks for your reply

**Comment by trackingallday** [comment]
Same issue in New Zealand +64

**Comment by hwschuur** [comment]
Similar issue from The Netherlands (+31). Username hwschuur

<img width="583" alt="image" src="https://github.com/keybase/keybase-issues/assets/6420430/8da2596d-1c09-4c8e-a652-4503a63f912e">

**Comment by moeffju** [comment]
Same from Germany (+49)

**Comment by cunneen** [comment]
Australia too (+61). Username `cunneen` .

**Comment by luxusko** [comment]
same in Slovakia (+421). Username `luxusko`.

**Comment by SP109** [comment]
Same problem with me, too.

username:SP109
Carrier: alditalk (ePlus,O2)

**Comment by josecelano** [comment]
Relates to:
- https://github.com/keybase/keybase-issues/issues/4259
- https://github.com/keybase/keybase-issues/issues/4194
- https://github.com/keybase/keybase-issues/issues/4182
- https://github.com/keybase/keybase-issues/issues/3646
- https://github.com/keybase/keybase-issues/issues/3502

**Comment by josecelano** [comment]
Same in Spain (+34)

**Comment by amy600-git** [comment]
> Same in Spain (+34)

Same in Germany (+49)

**Comment by gertzakis** [comment]
Same in Greece (+30)
